From 5c922739aeb5656d9110b02519d3c4d26013c5a7 Mon Sep 17 00:00:00 2001
From: Michael Trensch <mtrensch@hilscher.com>
Date: Mon, 10 May 2021 10:01:07 +0200
Subject: [PATCH 02/27] pl011: Fix invalid byte access in debug mode

---
 arch/arm/include/debug/pl01x.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/include/debug/pl01x.S b/arch/arm/include/debug/pl01x.S
index c7e02d0628bf..3eb29b3ef4ec 100644
--- a/arch/arm/include/debug/pl01x.S
+++ b/arch/arm/include/debug/pl01x.S
@@ -16,7 +16,7 @@
 #endif
 
 		.macro	senduart,rd,rx
-		strb	\rd, [\rx, #UART01x_DR]
+		strh	\rd, [\rx, #UART01x_DR]
 		.endm
 
 		.macro	waituartcts,rd,rx
-- 
2.25.1

