From 1636ac35f97c4602f18f70d97b2bee6f999ca6ef Mon Sep 17 00:00:00 2001
From: Michael Trensch <mtrensch@hilscher.com>
Date: Mon, 10 May 2021 10:01:07 +0200
Subject: [PATCH 02/25] pl011: Fix invalid byte access in debug mode

---
 arch/arm/include/debug/pl01x.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/include/debug/pl01x.S b/arch/arm/include/debug/pl01x.S
index 0c7bfa4c10db..e9dd76ca40f1 100644
--- a/arch/arm/include/debug/pl01x.S
+++ b/arch/arm/include/debug/pl01x.S
@@ -23,7 +23,7 @@
 #endif
 
 		.macro	senduart,rd,rx
-		strb	\rd, [\rx, #UART01x_DR]
+		strh	\rd, [\rx, #UART01x_DR]
 		.endm
 
 		.macro	waituartcts,rd,rx
-- 
2.25.1

