From cb152667b9390a141ba79185f5916ea88b42bb4b Mon Sep 17 00:00:00 2001
From: Michael Trensch <mtrensch@hilscher.com>
Date: Mon, 13 Feb 2023 17:11:07 +0100
Subject: [PATCH 8/8] pl022: Fix redefinition of flush

---
 drivers/spi/pl022_spi.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/spi/pl022_spi.c b/drivers/spi/pl022_spi.c
index 828eab3d34..c8923c91bd 100644
--- a/drivers/spi/pl022_spi.c
+++ b/drivers/spi/pl022_spi.c
@@ -107,7 +107,7 @@ static int pl022_spi_probe(struct udevice *bus)
 	return 0;
 }
 
-static void flush(struct pl022_spi_slave *ps)
+static void pl022_flush(struct pl022_spi_slave *ps)
 {
 	do {
 		while (readw(ps->base + SSP_SR) & SSP_SR_MASK_RNE)
@@ -126,7 +126,7 @@ static int pl022_spi_claim_bus(struct udevice *dev)
 	reg |= SSP_CR1_MASK_SSE;
 	writew(reg, ps->base + SSP_CR1);
 
-	flush(ps);
+	pl022_flush(ps);
 
 	return 0;
 }
@@ -137,7 +137,7 @@ static int pl022_spi_release_bus(struct udevice *dev)
 	struct pl022_spi_slave *ps = dev_get_priv(bus);
 	u16 reg;
 
-	flush(ps);
+	pl022_flush(ps);
 
 	/* Disable the SPI hardware */
 	reg = readw(ps->base + SSP_CR1);
-- 
2.25.1

